servicetalk-grpc-health/src/main/java/io/servicetalk/grpc/health/DefaultHealthService.java

```diff
@@ -81,8 +82,8 @@ public DefaultHealthService(Predicate<String> watchAllowed) {
     public Single<HealthCheckResponse> check(final GrpcServiceContext ctx, final HealthCheckRequest request) {
         HealthValue health = serviceToStatusMap.get(request.getService());
         if (health == null) {
-            return Single.failed(new GrpcStatus(NOT_FOUND, null, "unknown service: " + request.getService())
-                    .asException());
+            return Single.failed(new GrpcStatusException(
+                    new GrpcStatus(NOT_FOUND, "unknown service: " + request.getService())));
```

Comment:
Could be a follow-up, but something to consider: for use-cases like this one it looks useful to be able to create a stackless `GrpcStatusException` because it's not used for anything else except population of the `grpc-status` & `grpc-message`.
WDYT about adding `GrpcStatusException(String message, Throwable cause, boolean enableSuppression, boolean writableStackTrace)` ctor? That way we and users can create static exceptions with no stacktrace and no suppressions.

Reply:
lets consider this as a followup if necessary.